logic/tool/TypeResolver.java

```diff
@@ -118,7 +146,8 @@ public Conjunction resolve(Conjunction conjunction) {
                         mapping.putIfAbsent(ref, new HashSet<>());
                         assert vertex.isType();
                         // TODO: This filter should not be needed if we enforce traversal only to return non-abstract
-                        if (!(vertex.asType().isAbstract() && traversalConstructor.getVariable(ref).isThing()))
+                        if (!traversalConstructor.getVariable(ref).isPresent()) return;
+                        if (!(vertex.asType().isAbstract() && traversalConstructor.getVariable(ref).get().isThing()))
```

Comment:
This might not be the neatest way, but this will be refactored in an upcoming PR anyway.